backend/btrixcloud/storages.py

```diff
@@ -24,462 +25,497 @@
     sync_get_log_stream,
 )
 
+from .utils import is_bool
+
 
 CHUNK_SIZE = 1024 * 256
 
 
 # ============================================================================
-def init_storages_api(org_ops, crawl_manager, user_dep):
-    """API for updating storage for an org"""
+class StorageOps:
+    """All storage handling, download/upload operations"""
 
-    router = org_ops.router
-    org_owner_dep = org_ops.org_owner_dep
+    def __init__(self, org_ops, crawl_manager):
+        self.storages = {}
+
+        self.org_ops = org_ops
+        self.crawl_manager = crawl_manager
+
+        self.is_local_minio = is_bool(os.environ.get("IS_LOCAL_MINIO"))
+
+        with open("/tmp/storages/storages.json", encoding="utf-8") as fh:
+            storage_list = json.loads(fh.read())
+
+        for storage in storage_list:
+            name = storage.get("name")
+            type_ = storage.get("type", "s3")
+            if type_ == "s3":
+                self.storages[name] = self._create_s3_storage(storage)
+            else:
+                # expand when additional storage options are supported
+                raise TypeError("Only s3 storage supported for now")
+
+    def _create_s3_storage(self, storage):
+        """create S3Storage object"""
+        endpoint_url = storage["endpoint_url"]
+        bucket_name = storage.get("bucket_name")
+        if bucket_name:
+            endpoint_url += bucket_name + "/"
```

Comment:
```suggestion
        endpoint_url = os.path.join(storage["endpoint_url"], "")
        bucket_name = storage.get("bucket_name")
        if bucket_name:
            endpoint_url = os.path.join(f"{endpoint_url}{bucket_name}", "")
```
Will ensure that `endpoint_url` always ends in a trailing slash, before and after adding bucket, regardless of whether the user included one in the storage config

Reply:
hm, maybe even cleaner would be (works with and without bucket_name, with single trailing slash)
```suggestion
        endpoint_url = storage["endpoint_url"]
        bucket_name = storage.get("bucket_name", "")
        endpoint_url = os.path.join(endpoint_url, bucket_name, "")
```

However, this was adopted from https://github.com/webrecorder/browsertrix-cloud/blob/main/chart/templates/secrets.yaml#L38 and that is still used to create the secrets,
so if endpoint_url there doesn't have a slash, it'll get incorrect result there.

Reply:
Leaving as is to match the behavior in helm chart for:
```
  {{- if $storage.bucket_name }}
  STORE_ENDPOINT_URL: "{{ $storage.endpoint_url }}{{ $storage.bucket_name }}/"
  {{- else }}
  STORE_ENDPOINT_URL: "{{ $storage.endpoint_url }}"
  {{- end }}
  ```